src/librustc_typeck/check/expr.rs

```diff
@@ -1440,6 +1440,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         display
     }
 
+    fn complain_about_invalid_type_params(&self, args: Option<Span>) {
```

Comment:
Let's move this to after `check_field` as the other error methods related to `check_field` also come after.